jrt/src/com/yahoo/jrt/Transport.java

```diff
@@ -380,6 +380,7 @@ private void run() {
         try { selector.close(); } catch (Exception e) {}
         closer.shutdown().join();
         connector.exit().join();
+        try { cryptoEngine.close(); } catch (Exception e) {}
```

Comment:
Correct. No usage like that as of now.
Any suggestions on how this should be solved.
Add a factory and always provide your own, or always provide from the outside.

Reply:
In C++ we have a get_default instead of 'create', which lets all the default engines be the same shared object (which is configured based on environment variables). Since static stuff is more frowned upon in Java, we might want to pass it in from the outside in order to share it. Even if it is a bit ugly, I think the balance between convenience and forceful correctness is to perform conditional close based on self-create or passed-in crypto engine.

Reply:
When you pass it in from the outside you can wrap it in a proxy with a noop close(), and handle the close on the outside.

Reply:
It's just wrong to call close on something which you didn't open yourself.

Reply:
Currently it is created by self. But there is an api that lets you do both, but that is not used for anything but testing different crypto engines. So this @bjorncs or @havardpe will fix properly when they allw for a single shared cryptoengine.